src/mmw/apps/geoprocessing_api/validation.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+from __future__ import unicode_literals
+from __future__ import division
+
+import json
+
+from numbers import Number
+
+from django.conf import settings
+from django.contrib.gis.geos import GEOSGeometry
+from rest_framework.exceptions import ValidationError
+
+
+def create_invalid_rwd_location_format_error_msg(loc):
+    return ('Invalid required `location` parameter value `{}`. Must be a '
+            '`[lat, lng] where `lat` and `lng` are numeric.'.format(loc))
+
+
+def validate_location_format(loc):
+    if loc is None or type(loc) is not list or len(loc) is not 2:
+        return False
+    else:
+        [lat, lng] = loc
+        return (isinstance(lat, Number) and isinstance(lng, Number) and
+                type(lat) is not bool and type(lng) is not bool)
+
+
+def create_invalid_rwd_data_source_error_msg(source):
+    return ('Invalid optional `dataSource` parameter value `{}`. Must be '
+            '`drb` or `nhd`.'.format(source))
+
+
+def validate_rwd_data_source(source):
+    return source in ['drb', 'nhd']
+
+
+def create_invalid_rwd_snapping_error_msg(snapping):
+    return ('Invalid optional `snappingOn` parameter value `{}`. Must be '
+            '`true` or `false`.'.format(snapping))
+
+
+def create_invalid_rwd_simplify_param_type_error_msg(simplify):
+    return ('Invalid optional `simplify` parameter value: `{}`. Must be a '
+            'number.'.format(simplify))
+
+
+def validate_rwd_simplify_param_type(simplify):
+    return isinstance(simplify, Number) or simplify is False
+
+
+def get_aoi_sq_km(aoi):
+    geom = GEOSGeometry(aoi, 4326).transform(5070, clone=True)
+    return geom.area / 1000000
+
+
+def create_excessive_aoi_size_error_msg(aoi):
+    return ('Area of interest is too exceeds maximum size: submitted {} sq km '
+            'but the maximum size is {}'.format(get_aoi_sq_km(aoi),
+                                                settings.MMW_MAX_AREA))
+
+
+def validate_analyze_aoi_size_below_max_area(aoi):
+    return get_aoi_sq_km(aoi) < settings.MMW_MAX_AREA
+
+
+def create_invalid_shape_error_msg(aoi):
+    return ('Area of interest is invalid: {}'.format(
+            GEOSGeometry(aoi, 4326).valid_reason))
+
+
+def validate_aoi_does_not_self_intersect(aoi):
```

Comment:
Consider renaming these to some other verb like `check` to reserve `validate` only for functions in this file that raise proper `ValidationErrors`. Alternatively, move the `ValidationError` in `validate_aoi` here.